src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -75,61 +82,76 @@ public GlobbingUrlBuilder(IFileProvider fileProvider, IMemoryCache cache, PathSt
         /// <returns>The list of URLs</returns>
         public virtual ICollection<string> BuildUrlList(string staticUrl, string includePattern, string excludePattern)
         {
-            var urls = new HashSet<string>(StringComparer.Ordinal);
```

Comment:
HashSet has no guarantee of ordering (which we expect from this result). It just happened to keep the order in which items were inserted.